riscv/triggers.cc

```diff
@@ -17,6 +17,94 @@ action_t trigger_t::legalize_action(reg_t val) const noexcept {
   return (val > ACTION_MAXVAL || (val == ACTION_DEBUG_MODE && get_dmode() == 0)) ? ACTION_DEBUG_EXCEPTION : (action_t)val;
 }
 
+reg_t trigger_t::tdata3_read(const processor_t * const proc) const noexcept {
+  auto xlen = proc->get_xlen();
+  reg_t tdata3 = 0;
+  if (xlen == 32) {
+    tdata3 = set_field(tdata3, CSR_TEXTRA32_MHVALUE, mhvalue);
+    tdata3 = set_field(tdata3, CSR_TEXTRA32_MHSELECT, mhselect);
+    tdata3 = set_field(tdata3, CSR_TEXTRA32_SBYTEMASK, sbytemask);
+    tdata3 = set_field(tdata3, CSR_TEXTRA32_SVALUE, svalue);
+    tdata3 = set_field(tdata3, CSR_TEXTRA32_SSELECT, sselect);
+  } else {
+    tdata3 = set_field(tdata3, CSR_TEXTRA64_MHVALUE, mhvalue);
+    tdata3 = set_field(tdata3, CSR_TEXTRA64_MHSELECT, mhselect);
+    tdata3 = set_field(tdata3, CSR_TEXTRA64_SBYTEMASK, sbytemask);
+    tdata3 = set_field(tdata3, CSR_TEXTRA64_SVALUE, svalue);
+    tdata3 = set_field(tdata3, CSR_TEXTRA64_SSELECT, sselect);
+  }
```

Comment:
Add additonal macros to ```trigger.h```.
@timsifive do you think those macro should be somewhere else? ```debug_defines.h```? or?

Reply:
Everything in debug_defines.h is auto-generated from the same XML that generates the debug spec registers itself. That mechanism supports some basic operations based on XLEN, but using `? :` like is needed here won't work well. (At the very least, it will be hard to comprehend the document, which is the primary goal.)

I suggest putting it in `triggers.h`. These values won't change in the future.